### PR TITLE
Add bonus and rate

### PR DESCRIPTION
**Pull request by rastorc3v**
- Add styles to bonus component (not responsive)
- Add rate component 
- Add animation (still not good)

### CODE REVIEW

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BonusComponent implements OnInit {
+  @Input() bonusButtonClick: () => void;
+  @Input() bonus: IBonus;
+
+  public user: IUser;
+  public isModeratorOrAdmin = false;
+  public isForm = false;
+  public loadingProcess = {
```
Comment (rmurtazin) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.ts:21]: I can't realize a reason of this object

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  public rate(rating: number): void {
+    this.loadingProcess.start = true;
+    const bonusRateSubscription = this.bonusesService
+      .rate(this.bonus.id, Math.floor(rating / 10))
+      .subscribe((newBonus: IBonus) => {
+        this.bonus.rating = newBonus.rating;
+        this.successRate();
+        bonusRateSubscription.unsubscribe();
```
Comment (rmurtazin) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.ts:58]: why do you unsubscribe here after getting one portion of data? 
If it really needed you can use 'take(1)' operator of rxjs

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  public onBonusButtonClick(): void {
+    this.bonusButtonClick();
+  }
+
+  public rate(rating: number): void {
+    this.loadingProcess.start = true;
+    const bonusRateSubscription = this.bonusesService
+      .rate(this.bonus.id, Math.floor(rating / 10))
+      .subscribe((newBonus: IBonus) => {
+        this.bonus.rating = newBonus.rating;
```
Comment (rmurtazin) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.ts:56]: Do you think is it immutable obj?

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        this.successRate();
+        bonusRateSubscription.unsubscribe();
+      });
+  }
+
+  public successRate(): void {
+    this.loadingProcess.endLoading();
+    // if we set default change detection strategy on stars component we can remove next line
+    this.changeDetection.detectChanges();
```
Comment (rmurtazin) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.ts:65]: remove it please

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    setTimeout(() => {
+      this.loadingProcess.reset();
+      this.isForm = false;
+      this.changeDetection.detectChanges();
```
Comment (rmurtazin) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.ts:69]: and here

File: src/app/shared/components/rating/rate/rate.component.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+@Component({
+  selector: 'app-rate',
+  templateUrl: './rate.component.html',
+  styleUrls: ['./rate.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RateComponent implements OnInit {
+  @Input() isForm: boolean;
+  @Output() ratingWasChanged = new EventEmitter<number>();
+  @Input() bonus: IBonus;
```
Comment (rmurtazin) [inline src/app/shared/components/rating/rate/rate.component.ts:21]: try to keep an order of properties by type

File: src/app/shared/components/rating/rate/rate.component.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  ngOnInit(): void {
+    this.startPosition = this.bonus.rating * 10;
+  }
+
+  public onMatSliderChange(slider: MatSliderChange): void {
+    this.bonus.rating = slider.value / 10;
+  }
+
+  public rate(): void {
```
Comment (rmurtazin) [inline src/app/shared/components/rating/rate/rate.component.ts:35]: the name of method doesn't explain it's functional

File: src/app/shared/components/rating/rating.module.ts [hunk 8]
```diff
@@ -1,10 +1,15 @@
...
+import { MatButtonModule } from '@angular/material/button';
+import { MatSliderModule } from '@angular/material/slider';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 
 @NgModule({
-  declarations: [StarsComponent],
-  imports: [MatIconModule],
-  exports: [StarsComponent],
+  declarations: [StarsComponent, RateComponent],
+    imports: [MatIconModule, CommonModule, MatButtonModule, MatSliderModule, MatProgressSpinnerModule],
```
Comment (rmurtazin) [inline src/app/shared/components/rating/rating.module.ts:12]: move each module to new string, follow mutual flow of code style please

File: src/app/shared/components/rating/rate/rate.component.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+  public submitRating(): void {
+    this.disableButton = true;
+    this.animationStart = true;
+    this.bonusesService
+      .rate(this.bonus.id, Math.floor(this.bonus.rating / 10))
+      .pipe(take(1))
+      .subscribe((newBonus: IBonus) => {
+        this.animationStart = false;
+        this.toasterService.showSuccess('Your review has been added', 'Rated');
```
Comment (umilanovich) [inline src/app/shared/components/rating/rate/rate.component.ts:48]: Can we move message to translation file?

File: src/app/shared/components/bonus-list-container/bonus/bonus.component.scss [hunk 10]
```diff
@@ -0,0 +1,205 @@
+@import 'src/styles/variables';
+
```
Comment (umilanovich) [inline src/app/shared/components/bonus-list-container/bonus/bonus.component.scss:2]: Please remove unnecessary spaces. One is enough

**Review by umilanovich — APPROVED** [review]

**Pull request merged**